api/src/services/application.service.ts

```diff
@@ -625,18 +655,39 @@ export class ApplicationService {
         id: dto.listings.id,
       },
       include: {
-        jurisdictions: true,
-        // support unit group availability logic in email
-        unitGroups: true,
-        // multiselect questions and address is needed for geocoding
+        jurisdictions: { include: { featureFlags: true } },
+        // address is needed for geocoding
+        listingsBuildingAddress: true,
         listingMultiselectQuestions: {
-          include: {
-            multiselectQuestions: true,
-          },
+          include: { multiselectQuestions: true },
         },
-        listingsBuildingAddress: true,
+        // support unit group availability logic in email
+        unitGroups: true,
       },
     });
+
+    const enableV2MSQ = doJurisdictionHaveFeatureFlagSet(
+      listing?.jurisdictions as unknown as Jurisdiction,
+      FeatureFlagEnum.enableV2MSQ,
+    );
+
+    if (enableV2MSQ) {
+      const listingMultiselectIds = listing.listingMultiselectQuestions.map(
+        (msq) => {
+          return msq.multiselectQuestionId;
+        },
+      );
+      if (
+        !dto.applicationSelections.every(({ multiselectQuestion }) => {
+          return listingMultiselectIds.includes(multiselectQuestion.id);
+        })
```

Comment:
nit: I'm ok keeping this as is if you think that the way you have it here is more readable. However, you could do the inverse here by using the `some` function.
```
 dto.applicationSelections.some(({ multiselectQuestion }) => {
          return !listingMultiselectIds.includes(multiselectQuestion.id);
        })
```
Either way is just as performant because they both run until they hit a record that causes the function to fail.